### PR TITLE
Configure CosmosDB SDK to serialize JSON with camel case

### PR DESCRIPTION
**Pull request by TaoChenOSU**
### Motivation and Context

The CosmosDb SDK serializes entities into Json using the same property names. We would like it to serialize the entities to Json with the property names in camel case.

### Description

Configure the CosmosDb client to use camel case for serialization.

### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [ ] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

**Comment by dluc** [comment]
@TaoChenOSU could you elaborate why `System.Text.Json` doesn't work?

File: samples/apps/copilot-chat-app/webapi/Storage/CosmosDbContext.cs [hunk 1]
```diff
@@ -29,7 +29,7 @@ public CosmosDbContext(string connectionString, string database, string containe
     }
 
     /// <inheritdoc/>
-    public IQueryable<T> QueryableEntities => this._container.GetItemLinqQueryable<T>().AsQueryable();
+    public IQueryable<T> QueryableEntities => this._container.GetItemLinqQueryable<T>(allowSynchronousQueryExecution: true).AsQueryable();
```
Comment (dluc) [inline samples/apps/copilot-chat-app/webapi/Storage/CosmosDbContext.cs:32]: is this change in scope?

Comment (TaoChenOSU) [reply samples/apps/copilot-chat-app/webapi/Storage/CosmosDbContext.cs]: Good catch. Changed it to async.

**Comment by TaoChenOSU** [comment]
> @TaoChenOSU could you elaborate why `System.Text.Json` doesn't work?

Microsoft.Azure.Cosmos v3 takes dependency on Json.NET.

**Review by adrianwyatt — APPROVED** [review]
